File: code/java/validPalindrome.java
class Solution {
    public boolean validPalindrome(String s) {
        if(s.length() == 0)
            return true;
        s = s.toLowerCase();
        int head = 0;
        int tail = s.length() - 1;
        while(head < tail){
            if(helper(s.charAt(head)) == false){
                head++;
                continue;
            }
            if(helper(s.charAt(tail)) == false){
                tail--;
                continue;
            }
            if(s.charAt(head) != s.charAt(tail))
                return false;
            head++;
            tail--;
        }

        return true;     
    }
    
    public boolean helper(char c){
        if((c >= '0' && c<= '9') || (c >= 'a'  && c<= 'z'))
            return true;
        else
            return false;
    }
}

File: code/python/validPalindrome.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        if len(s) == 0:
            return True
        s = s.lower()
        s1 = ""
        s2 = ""
        for i in range(len(s)):
            if (s[i] >= 'a' and s[i] <= 'z') or (s[i] >= '0' and s[i] <= '9'):
                s1 = s[i] + s1
                s2 = s2 + s[i]
        return s1==s2
